extensions/http-curl/client/HTTPCallback.h

```diff
@@ -78,7 +78,7 @@ class HttpStreamingCallback : public ByteInputCallBack {
 
   }
 
-  virtual int64_t process(uint8_t *vector, size_t size) {
+  virtual int64_t process(const uint8_t *vector, size_t size) {
```

Comment:
in general when i see something like const uint8_t *ptr, then I wonder what the true intent may be there. You have a pointer that can be modified. Should also make that pointer const too, but as with all comments I'm not sure that's the best approach here.